### PR TITLE
fix memory leak 

### PR DESCRIPTION
**Pull request by aloyszhang**
### Motivation
This pr fix memory leak by release bytebuf.

### CODE REVIEW

**Review by codelipenghui — APPROVED** [review]

**Review by hangc0276 — APPROVED** [review]

**Review by 315157973 — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/intercept/ManagedLedgerInterceptorImpl.java [hunk 1]
```diff
@@ -99,6 +100,12 @@ public void onManagedLedgerLastLedgerInitialize(String name, LedgerHandle lh) {
             }
         } catch (org.apache.bookkeeper.client.api.BKException | InterruptedException e) {
             log.error("[{}] Read last entry error.", name, e);
+        } finally {
+            if (ledgerEntries != null) {
```
Comment (sijie) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/intercept/ManagedLedgerInterceptorImpl.java:104]: LedgerEntries has a `close()` interface.

**Review by sijie — CHANGES_REQUESTED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/intercept/ManagedLedgerInterceptorImpl.java [hunk 2]
```diff
@@ -81,10 +81,11 @@ public void onManagedLedgerPropertiesInitialize(Map<String, String> propertiesMa
 
     @Override
     public void onManagedLedgerLastLedgerInitialize(String name, LedgerHandle lh) {
+        LedgerEntries ledgerEntries = null;
         try {
             for (BrokerEntryMetadataInterceptor interceptor : brokerEntryMetadataInterceptors) {
                 if (interceptor instanceof AppendIndexMetadataInterceptor && lh.getLastAddConfirmed() >= 0) {
-                    LedgerEntries ledgerEntries =
+                    ledgerEntries =
```
Comment (eolivelli) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/intercept/ManagedLedgerInterceptorImpl.java:88]: What happens if we get to here multiple times?
There is no invariant that enforces the presence of only one AppendIndexMetadataInterceptor

Comment (eolivelli) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/intercept/ManagedLedgerInterceptorImpl.java:88]: You can use a try-with-resources block and close the object as soon as we do not it anymore

**Review by eolivelli — CHANGES_REQUESTED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/intercept/ManagedLedgerInterceptorImpl.java (hunk 2 above)
Comment (aloyszhang) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/intercept/ManagedLedgerInterceptorImpl.java:88]: There will be only on `AppendIndexMetadataInterceptor` for each `ManagedLedger`

**Review by sijie — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/intercept/ManagedLedgerInterceptorImpl.java (hunk 2 above)
Comment (eolivelli) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/intercept/ManagedLedgerInterceptorImpl.java:88]: what about
```
try (LedgerEntries ledgerEntries = lh.read....) {
...
}
```

Comment (aloyszhang) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/intercept/ManagedLedgerInterceptorImpl.java:88]: try-with-resources is alternative way and makes code more tidy. 
 I think it's better to raise an individual issue for refactor code like try-with-resources

**Pull request merged**